### PR TITLE
Controlled sysctl changes for Elasticsearch5

### PR DESCRIPTION
**Pull request by ewolinetz**
As required for ES5 to run, we need a minimum `vm.max_map_count` value of `262144`.
ES5 is scheduled for 3.11.

This change adds: 
* Updating to require es node selectors for es5 install
* Only create sysctl files for nodes es will run on 
* Clean up sysctl files when uninstalling logging

Addresses concerns in https://github.com/openshift/origin/pull/19823#issuecomment-403860504

### CODE REVIEW

File: playbooks/openshift-logging/private/config.yml [hunk 1]
```diff
@@ -13,33 +13,93 @@
...
+      - name: Ensure that ElasticSearch Ops has nodes to run on
+        import_role:
+          name: openshift_control_plane
+          tasks_from: ensure_nodes_matching_selector.yml
+        vars:
+          openshift_master_ensure_nodes_selector: "{{ openshift_logging_es_ops_nodeselector | map_to_pairs }}"
+          openshift_master_ensure_nodes_service: Elasticsearch
+
+      - command: >
+          oc get nodes -l {{ openshift_logging_es_ops_nodeselector | map_to_pairs }} -o jsonpath={.items[*].metadata.name}
```
Comment (ewolinetz) [inline playbooks/openshift-logging/private/config.yml]: I need to make these follow the current patterns for using OC

File: playbooks/openshift-logging/private/config.yml [hunk 2]
```diff
@@ -13,33 +13,93 @@
...
+      tasks_from: ensure_nodes_matching_selector.yml
+    vars:
+      openshift_master_ensure_nodes_selector: "{{ openshift_logging_es_nodeselector | map_to_pairs }}"
+      openshift_master_ensure_nodes_service: Elasticsearch
+
+  - command: >
+      oc get nodes -l {{ openshift_logging_es_nodeselector | map_to_pairs }} -o jsonpath={.items[*].metadata.name}
+    register: openshift_logging_es_hosts
+
+  - when: openshift_logging_use_ops | default(false) | bool
```
Comment (richm) [inline playbooks/openshift-logging/private/config.yml]: Do we need ops for 3.11/master branch?

File: playbooks/openshift-logging/private/config.yml [hunk 3]
```diff
@@ -13,33 +13,93 @@
...
+  - name: Ensure that ElasticSearch has nodes to run on
+    import_role:
+      name: openshift_control_plane
+      tasks_from: ensure_nodes_matching_selector.yml
+    vars:
+      openshift_master_ensure_nodes_selector: "{{ openshift_logging_es_nodeselector | map_to_pairs }}"
+      openshift_master_ensure_nodes_service: Elasticsearch
+
+  - command: >
+      oc get nodes -l {{ openshift_logging_es_nodeselector | map_to_pairs }} -o jsonpath={.items[*].metadata.name}
```
Comment (ewolinetz) [inline playbooks/openshift-logging/private/config.yml]: I need to make these follow the current patterns for using OC

File: playbooks/openshift-logging/private/config.yml (hunk 2 above)
Comment (ewolinetz) [reply playbooks/openshift-logging/private/config.yml]: Currently we haven't deprecated the use of that variable, so for the time being, yes.

File: playbooks/openshift-logging/private/config.yml [hunk 4]
```diff
@@ -13,33 +13,101 @@
           status: "In Progress"
           start: "{{ lookup('pipe', 'date +%Y%m%d%H%M%SZ') }}"
 
+- name: Verify and collect ES hosts
```
Comment (richm) [inline playbooks/openshift-logging/private/config.yml]: I think all of these elasticsearch/logging tasks should be in a `when: openshift_logging_install_logging | default(false) | bool` block.

Comment (ewolinetz) [reply playbooks/openshift-logging/private/config.yml]: Should we then just iterate over `all` hosts for the removal case?

File: playbooks/openshift-logging/private/config.yml [hunk 5]
```diff
@@ -13,33 +13,101 @@
...
+          path: /etc/sysctl.d/99-elasticsearch.conf
+        register: _99_es_conf
+
+      - name: Check for current value of vm.max_map_count in 99-elasticsearch.conf
+        command: >
+          sed /etc/sysctl.d/99-elasticsearch.conf -e 's/vm.max_map_count=\(.*\)/\1/'
+        register: _curr_vm_max_map_count
+        when: _99_es_conf.stat.exists
+
+      - name: Updating vm.max_map_count value
```
Comment (richm) [inline playbooks/openshift-logging/private/config.yml]: Do you need the above checking, or can you just use the sysctl module?  I think the sysctl module is smart enough to set the setting if not set or not the current value, and reload if changed.

**Comment by ewolinetz** [comment]
/hold
I want to give enough people on this a chance to have concerns addressed, I'll remove hold by noon (eastern) tomorrow

**Comment by ewolinetz** [comment]
Logging tests need to be updated so that we are specifying a node selector and also labelling our nodes as part of doing an initial installation of OCP

**Comment by sdodson** [comment]
Seems like there's some quoting problems in the logging test job, maybe remove ' ' or use double quotes for the key and value inside the single quotes?
```
-e 'openshift_logging_es_nodeselector={node-role.kubernetes.io/infra:' 'true}' -e 'openshift_logging_es_ops_nodeselector={node-role.kubernetes.io/infra:' 'true}'
```

**Comment by ewolinetz** [comment]
@sdodson yeah, i'm not sure why its looking like that, the changes and the gen'd files don't look like that...
https://github.com/openshift/aos-cd-jobs/pull/1519/files

There should not be an outer single quote there at all

**Comment by ewolinetz** [comment]
/test logging

none of the other logging tests are failing in the same place...

**Comment by ewolinetz** [comment]
known fluentd flake
/test logging

**Pull request merged**